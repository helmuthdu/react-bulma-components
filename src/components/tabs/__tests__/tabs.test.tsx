import { render } from '@testing-library/react';
import * as React from 'react';
import { Tab, Tabs } from '..';

describe('Tabs component', () => {
  // @ts-ignore
  it.each([[Tabs], [Tab]])('should render', Component => {
    const { asFragment } = render(<Component>Content</Component>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Tabs className="other-class test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render as an html section', () => {
    const { asFragment } = render(<Tabs as="section">This should be a section</Tabs>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have custom inline styles', () => {
    const { asFragment } = render(
      <Tabs as="section" style={{ width: 200, zIndex: 1 }}>
        This should be a section with custom styles
      </Tabs>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component with different tag', () => {
    // eslint-disable-next-line react/prop-types
    const Custom = ({ children, ...props }: any) => (
      <li {...props}>
        Custom
        {children}
      </li>
    );
    const { asFragment } = render(<Tabs as={Custom}>This should be a li element</Tabs>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render propertly Inside Tabs', () => {
    const { asFragment } = render(
      <Tabs>
        <Tab>Tab 1</Tab>
        <Tab>Tab 2</Tab>
        <Tab>Tab 3</Tab>
      </Tabs>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
