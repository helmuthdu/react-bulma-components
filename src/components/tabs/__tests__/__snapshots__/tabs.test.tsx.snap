// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component should render component with different tag 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <li
      class="tabs"
    >
      Custom
      <ul>
        This should be a li element
      </ul>
    </li>
  </div>
</DocumentFragment>
`;

exports[`Tabs component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <nav
      class="tabs other-class test"
    >
      <ul />
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Tabs component should have custom inline styles 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <section
      class="tabs"
      style="width: 200px; z-index: 1;"
    >
      <ul>
        This should be a section with custom styles
      </ul>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Tabs component should render 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <nav
      class="tabs"
    >
      <ul>
        Content
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Tabs component should render 2`] = `
<DocumentFragment>
  <li
    class=""
  >
    <a>
      Content
    </a>
  </li>
</DocumentFragment>
`;

exports[`Tabs component should render as an html section 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <section
      class="tabs"
    >
      <ul>
        This should be a section
      </ul>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Tabs component should render propertly Inside Tabs 1`] = `
<DocumentFragment>
  <div
    class="tabs-container"
  >
    <nav
      class="tabs"
    >
      <ul>
        <li
          class=""
        >
          <a>
            Tab 1
          </a>
        </li>
        <li
          class=""
        >
          <a>
            Tab 2
          </a>
        </li>
        <li
          class=""
        >
          <a>
            Tab 3
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
